client/packages/host/src/Admin/SyncSettings.tsx

```diff
@@ -92,56 +38,62 @@ const SyncSettingsForm = ({
   isDisabled: boolean;
   isSaving: boolean;
   isValid: boolean;
-  settings?: SyncSettings;
+  settings: SyncSettingsInput;
   onSave: () => void;
-  setSyncSettings: (syncSettings: SyncSettings) => void;
+  setSyncSettings: (syncSettings: SyncSettingsInput) => void;
 }) => {
   const t = useTranslation('common');
+  const setSettings = (
+    property: keyof SyncSettingsInput,
+    value: number | string
+  ) => setSyncSettings({ ...settings, [property]: value });
+
+  const { url, username, password, intervalSeconds } = settings;
   return (
     <form style={{ width: '100%' }}>
       <Setting
         title={t('label.settings-url')}
         component={
-          <StringSyncSetting
-            property="url"
-            settings={settings}
+          <BasicTextInput
+            value={url}
+            onChange={e => setSettings('url', e.target.value)}
+            type={'text'}
             disabled={isDisabled}
-            update={setSyncSettings}
           />
         }
       />
       <Setting
         title={t('label.settings-username')}
         component={
-          <StringSyncSetting
-            property="username"
-            settings={settings}
+          <BasicTextInput
+            value={username}
+            onChange={e => setSettings('username', e.target.value)}
+            type={'text'}
             disabled={isDisabled}
-            update={setSyncSettings}
           />
         }
       />
       <Setting
         title={t('label.settings-password')}
         component={
-          <StringSyncSetting
-            property="password"
-            type="password"
-            autocomplete="sync-password"
-            settings={settings}
+          <BasicTextInput
+            value={password}
+            inputProps={{ autocomplete: 'sync-password' }}
+            onChange={e => setSettings('password', e.target.value)}
+            type={'password'}
             disabled={isDisabled}
-            update={setSyncSettings}
           />
         }
       />
       <Setting
         title={t('label.settings-interval')}
         component={
-          <NumericSyncSetting
-            property="intervalSeconds"
-            settings={settings}
+          <NumericTextInput
+            value={intervalSeconds}
+            onChange={e =>
```

Comment:
I have wondered about changing the numeric text input so that the onChange returns an actual number. Seems silly that we do the event handling & number conversion all over the place 🤔 

```
export interface NumericTextInputProps
  extends Omit<StandardTextFieldProps, 'onChange'> {
  onChange: (value: number) => void;
  width?: number;
}

export const NumericTextInput: FC<NumericTextInputProps> = React.forwardRef(
  ({ sx, InputProps, width = 75, onChange, ...props }, ref) => (
    <BasicTextInput
      ref={ref}
      sx={{
        '& .MuiInput-input': { textAlign: 'right', width: `${width}px` },
        ...sx,
      }}
      InputProps={InputProps}
      onChange={e => onChange(Number(e.target.value))}
      type="number"
      {...props}
    />
  )
);
```

and 

```
<NumericTextInput
  value={intervalSeconds}
  onChange={seconds => setSettings('intervalSeconds', seconds)}
  disabled={isDisabled} 
/>
```

🤷

Reply:
Sounds good, added to carry over: https://github.com/openmsupply/open-msupply/issues/688